deployment/helm/node-feature-discovery/templates/worker.yaml

```diff
@@ -5,6 +5,10 @@ metadata:
   labels:
     {{- include "node-feature-discovery.labels" . | nindent 4 }}
     role: worker
+{{- with .Values.worker.daemonsetAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+{{- end }}
```

Comment:
```suggestion
  annotations:
    {{- toYaml .Values.worker.daemonsetAnnotations | nindent 4 }}
```

Reply:
As you want, the `with` will just not print the annotations if the dict is empty

Reply:
I just find the template easier to read this way. And, I don't think the output really matters as it's normally "machine interpreted" anyway